### PR TITLE
✨Added shop

### PR DESCRIPTION
**Pull request by davey2206**
## Content

Added the meta progression shop.

### How to use
Creating shop Item:
ShopItem
After this add it to the list of items in the savefile.

savefile

Creating shop:
ShopPrefab

note : the following files have public values to allow them to be save into a json file.

- `Assets/Scripts/MetaProgression/SaveFileObject.cs`
- `Assets/Scripts/MetaProgression/ShopAbility.cs`
- `Assets/Scripts/MetaProgression/ShopStat.cs`

### Added
`Assets/Scripts/MetaProgression/Shop/RogueApeStudio.Crusader.MetaProgression.Shop.asmdef` : was created
`Assets/Prefabs/UI/Shop.prefab` : was created
`Assets/Scripts/MetaProgression/Shop/Shop.cs` : was created
`Assets/Scripts/MetaProgression/ShopAbility.cs` : was created
`Assets/Scripts/MetaProgression/ShopAbilityObject.cs` : was created
`Assets/Scripts/MetaProgression/ShopItemObject.cs` : was created
`Assets/Scripts/MetaProgression/ShopStat.cs` : was created
`Assets/Scripts/MetaProgression/ShopStatObject.cs` : was created
`Assets/Scripts/MetaProgression/StatType.cs` : was created
`Assets/Scripts/MetaProgression/Shop/ShopItems` was created with all the ScriptableObjects for the shop items within

### Updated
`Assets/Scripts/SaveSystem→ Assets/Scripts/MetaProgression` : was  renamed
`Assets/Scripts/MetaProgression/RogueApeStudio.Crusader.SaveSystem.asmdef → Assets/Scripts/MetaProgression/RogueApeStudio.Crusader.MetaProgression.asmdef` :  was renamed
`Assets/Scenes/Menu/MainMenu.unity` : was updated
`Assets/Scripts/MetaProgression/SaveSystem.cs` : was updated
`Assets/Scripts/MetaProgression/SaveFile.asset` : was updated
`Assets/Scripts/MetaProgression/SaveFileObject.cs` : was updated
`Assets/Scripts/MetaProgression/SaveSystem.cs` : was updated
`Assets/Scripts/UI/MainMenu.cs` : was updated
`Assets/TextMesh Pro/Fonts/MedievalSharp SDF.asset` : was updated

### CODE REVIEW

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopBeam.asset
Comment (AoSkye29) [inline Assets/Scripts/MetaProgression/Shop/ShopItems/ShopBeam.asset:1]: Spelling mistake at 'destroys all your eremies' -> should be 'destroys all your enemies'

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopCavalryCharge.asset
Comment (AoSkye29) [inline Assets/Scripts/MetaProgression/Shop/ShopItems/ShopCavalryCharge.asset:1]: Spelling mistake at 'Summon allies that charge forwards damaging all enemies in there way' -> 'Summon allies that charge forward damaging all enemies in their path'.

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopChainAttack.asset
Comment (AoSkye29) [inline Assets/Scripts/MetaProgression/Shop/ShopItems/ShopChainAttack.asset:1]: The amount of bounces should be mentioned in description when we know a definite count after testing

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopCritRate.asset
Comment (AoSkye29) [inline Assets/Scripts/MetaProgression/Shop/ShopItems/ShopCritRate.asset:1]: Add the amount with which it increases by in description

Comment (AoSkye29) [reply Assets/Scripts/MetaProgression/Shop/ShopItems/ShopCritRate.asset:1]: *Do this for all stat increases

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopSpinny.asset
Comment (AoSkye29) [inline Assets/Scripts/MetaProgression/Shop/ShopItems/ShopSpinny.asset:1]: We need a proper name for this 
also small spelling mistake at 'arounds' -> should be 'around'

**Review by AoSkye29 — CHANGES_REQUESTED** [review]
Check comments for spelling errors that would need changing, otherwise looks good

File: Assets/Scripts/MetaProgression/Shop/ShopItems/ShopChainAttack.asset
Comment (davey2206) [reply Assets/Scripts/MetaProgression/Shop/ShopItems/ShopChainAttack.asset:1]: Will do

**Review by AoSkye29 — APPROVED** [review]

**Pull request merged**